scr/main/java/bittercode/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,193 @@
+                response = ResponseCode.SUCCESS.name();
+            }
+        } catch (Exception e) {
+            response += " : " + e.getMessage();
+            e.printStackTrace();
+        }
+        return response;
+    }
+
+    @Override
+    public String addBook(Book book) throws StoreException {
+        String responseCode = ResponseCode.FAILURE.name();
+        Connection con = DBUtil.getConnection();
+        try {
+            PreparedStatement ps = con.prepareStatement(addBookQuery);
+            ps.setString(1, book.getBarcode());
+            ps.setString(2, book.getName());
+            ps.setString(3, book.getAuthor());
+            ps.setDouble(4, book.getPrice());
+            ps.setInt(5, book.getQuantity());
+            int k = ps.executeUpdate();
+            if (k == 1) {
+                responseCode = ResponseCode.SUCCESS.name();
+            }
+        } catch (Exception e) {
+            responseCode += " : " + e.getMessage();
+            e.printStackTrace();
+        }
+        return responseCode;
+    }
+
+    @Override
+    public String updateBookQtyById(String bookId, int quantity) throws StoreException {
+        String responseCode = ResponseCode.FAILURE.name();
+        Connection con = DBUtil.getConnection();
+        try {
+            PreparedStatement ps = con.prepareStatement(updateBookQtyByIdQuery);
+            ps.setInt(1, quantity);
+            ps.setString(2, bookId);
+            ps.executeUpdate();
+            responseCode = ResponseCode.SUCCESS.name();
+        } catch (Exception e) {
+            responseCode += " : " + e.getMessage();
+            e.printStackTrace();
+        }
+        return responseCode;
+    }
+
+    @Override
+    public List<Book> getBooksByCommaSeperatedBookIds(String commaSeperatedBookIds) throws StoreException {
+        List<Book> books = new ArrayList<Book>();
+        Connection con = DBUtil.getConnection();
+        try {
+            String getBooksByCommaSeperatedBookIdsQuery = "SELECT * FROM " + BooksDBConstants.TABLE_BOOK
+                    + " WHERE " +
+                    BooksDBConstants.COLUMN_BARCODE + " IN ( " + commaSeperatedBookIds + " )";
+            PreparedStatement ps = con.prepareStatement(getBooksByCommaSeperatedBookIdsQuery);
+            ResultSet rs = ps.executeQuery();
+
+            while (rs.next()) {
+                String bCode = rs.getString(1);
+                String bName = rs.getString(2);
+                String bAuthor = rs.getString(3);
+                int bPrice = rs.getInt(4);
+                int bQty = rs.getInt(5);
+
+                Book book = new Book(bCode, bName, bAuthor, bPrice, bQty);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+
+        }
+        return books;
+    }
+
+    @Override
+    public String updateBook(Book book) throws StoreException {
+        String responseCode = ResponseCode.FAILURE.name();
+        Connection con = DBUtil.getConnection();
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Potential unhandled exception from getConnection</b></div>
<div id="fix">

It appears that the `DBUtil.getConnection()` method might throw a `StoreException`, but it's not being caught or handled here. Consider wrapping this call in a try-catch block to handle potential connection failures gracefully.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -174,8 +174,15 @@
      public String updateBook(Book book) throws StoreException {
          String responseCode = ResponseCode.FAILURE.name();
 -        Connection con = DBUtil.getConnection();
 -        try {
 +        Connection con = null;
 +        try {
 +            con = DBUtil.getConnection();
 +        } catch (StoreException e) {
 +            responseCode = ResponseCode.DATABASE_CONNECTION_FAILURE.name();
 +            return responseCode;
 +        }
 +
 +        try {
              PreparedStatement ps = con.prepareStatement(updateBookByIdQuery);
              ps.setString(1, book.getName());
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/3#issuecomment-2434572244>#7fa0a4</a></i></small>
</div>


---
🚩 Problem result:
- [ ] This issue was flagged incorrectly